tests/manager/core/test_coverage.py

```diff
@@ -546,9 +543,6 @@ def finalize_batch(self):
         # Two successes.
         assert (2, 0, 0) == counts
 
-        # finalize_batch() was called.
-        assert True == success_provider.finalized
-
```

Comment:
Seems like we would want to ensure that `finalize_batch` is called, since, in hindsight, it was easy to miss. Could you make a test that fails without the fixes you made above, but passes with them?